### PR TITLE
TRAINING PREP: Update complete branch react js

### PR DESCRIPTION
**Pull request by george-bafaloukas-forgerock**
Updating app to use the latest ASYNC config interface. 

Updating the JS SDK dependency

Adding README files

### CODE REVIEW

File: reactjs-todo/webpack.config.js [hunk 1]
```diff
@@ -8,16 +8,15 @@ module.exports = () => {
...
   const APP_URL = process.env.APP_URL || localEnv.APP_URL;
   const API_URL = process.env.API_URL || localEnv.API_URL;
   const DEBUGGER_OFF = process.env.DEBUGGER_OFF || localEnv.DEBUGGER_OFF;
   const DEVELOPMENT = process.env.DEVELOPMENT || localEnv.DEVELOPMENT;
   const JOURNEY_LOGIN = process.env.JOURNEY_LOGIN || localEnv.JOURNEY_LOGIN;
   const JOURNEY_REGISTER = process.env.JOURNEY_REGISTER || localEnv.JOURNEY_REGISTER;
   const WEB_OAUTH_CLIENT = process.env.WEB_OAUTH_CLIENT || localEnv.WEB_OAUTH_CLIENT;
-  const REALM_PATH = process.env.REALM_PATH || localEnv.REALM_PATH;
-
+  const WELLKNOWN_URL = process.env.WELLKNOWN_URL || localEnv.WELLKNOWN_URL;
```
Comment (ryanbas21) [inline reactjs-todo/webpack.config.js:18]: We probably need to remove REALM_PATH in the `main` branch as well, we should create something for this

File: reactjs-todo/webpack.config.js [hunk 2]
```diff
@@ -94,20 +93,20 @@ module.exports = () => {
       client: {
         overlay: false,
       },
-      port: 8443,
+      port: PORT,
```
Comment (ryanbas21) [inline reactjs-todo/webpack.config.js:96]: I don't think this is a problem in this repo - but if someone does put a `.env` file in the _root_ of the workspace, or at least - not in the project specific folder, technically its possible for this `PORT` variable to conflict with the `todo-api` `PORT` variable. I don't think they would get picked up but its something to consider.

We could avoid this by making the `todo-api` read something like `SERVER_PORT` instead to make it clear that these two `PORTS` are different

File: reactjs-todo/client/index.js [hunk 3]
```diff
@@ -21,16 +21,18 @@ import { AppContext, useGlobalStateMgmt } from './global-state';
...
+
+await Config.setAsync({
+  clientId: WEB_OAUTH_CLIENT, // e.g. PingOne Services Client GUID
+  redirectUri: `${window.location.origin}/callback`, // Redirect back to your app, e.g. 'https://localhost:8443/login?centralLogin=true' or the domain your app is served.
+  scope: "openid profile email address phone", // e.g. 'openid profile email address phone'
   serverConfig: {
-    baseUrl: AM_URL,
-    timeout: '5000',
+    wellknown: WELLKNOWN_URL,
+    timeout: '3000', // Any value between 3000 to 5000 is good, this impacts the redirect time to login. Change that according to your needs.
```
Comment (ryanbas21) [inline reactjs-todo/client/index.js:33]: I think timeout should be a number technically, right?

Comment (george-bafaloukas-forgerock) [reply reactjs-todo/client/index.js:33]: it works like this. TBH not changed that to anything in terms of type so I think it's OK to leave like this. On Main we have it as such

File: reactjs-todo/webpack.config.js (hunk 2 above)
Comment (george-bafaloukas-forgerock) [reply reactjs-todo/webpack.config.js:96]: yeah let me change that on the todo-api. I can see this confusing people

File: reactjs-todo/webpack.config.js (hunk 1 above)
Comment (george-bafaloukas-forgerock) [reply reactjs-todo/webpack.config.js:18]: yeah let's keep any main change is out of this scope. I will attend to the `start` branch after we are done on this.

We can keep an eye for MAIN improvements as part of another ticket

**Review by ryanbas21 — APPROVED** [review]

File: reactjs-todo/client/constants.js
Comment (rodrigoareis) [inline reactjs-todo/client/constants.js:1]: Update copyright

File: reactjs-todo/client/index.js
Comment (rodrigoareis) [inline reactjs-todo/client/index.js:1]: Update copyright

File: reactjs-todo/client/index.js (hunk 3 above)
Comment (cerebrl) [reply reactjs-todo/client/index.js:33]: Yeah, it should be a number, but JS will coerce the string to a number, if needed.

File: reactjs-todo/webpack.config.js
Comment (rodrigoareis) [inline reactjs-todo/webpack.config.js:1]: Missing copyright

File: todo-api/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# NodeJS Todo-API
+
+## Disclaimers
+
+This sample code is provided "as is" and is not a supported product of Ping. It's purpose is solely to demonstrate how the Ping JavaScript SDK can be implemented within a React application. Also, this is not a demonstration of React itself or instructional for _how_ to build a React app. There are many aspects to routing, state management, tooling and other aspects to building a React app that are outside of the scope of this project. For information about creating a React app, [visit React's official documentation](https://reactjs.org/docs/create-a-new-react-app.html).
```
Comment (rodrigoareis) [inline todo-api/README.md:5]: This app is vanilla JS, right?

File: reactjs-todo/client/index.js (hunk 3 above)
Comment (cerebrl) [reply reactjs-todo/client/index.js:33]: Actually, `timeout` isn't really needed anymore as it's totally optional. Even for Central Login, we don't rely on timeouts anymore if the user is not authenticated.

File: todo-api/src/app/constants.js
Comment (rodrigoareis) [inline todo-api/src/app/constants.js:1]: Update copyright

File: todo-api/src/index.js
Comment (rodrigoareis) [inline todo-api/src/index.js:1]: Update copyright

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Changes looks good, minor comments to address

File: todo-api/README.md (hunk 4 above)
Comment (cerebrl) [reply todo-api/README.md:5]: No, this is actually the Node.js app, but we should remove the React wording here as it's an API, so any client can use it, regardless of framework.

**Review by cerebrl — APPROVED** [review]

File: todo-api/README.md (hunk 4 above)
Comment (george-bafaloukas-forgerock) [reply todo-api/README.md:5]: This is being updated...forgot to remove the react bits and add the api. Shall be fixed soon...

**Review by rodrigoareis — APPROVED** [review]
Looks good to me

**Pull request merged**